test/tools/Modules/HelpersCommon/HelpersCommon.psm1

```diff
@@ -312,4 +313,29 @@ function Start-NativeExecution
     } finally {
         $script:ErrorActionPreference = $backupEAP
     }
+# Test if the scriptblock (cmdlet) can be stopped
+function Test-Stopping
+{
+    [CmdletBinding()]
+    param (
+        [ScriptBlock]$sb,
+        [int]$TimeoutInMilliseconds = 10000,
+        [int]$IntervalInMilliseconds = 1000
+        )
+
+    try {
+        $ps = [PowerShell]::Create()
+        $null = $ps.AddScript($sb)
+        Enable-Testhook 'ActivateSleepForStoppingTest'
+        $null = $ps.BeginInvoke()
+        Wait-UntilTrue { $ps.Streams.Verbose.Count -gt 0 } -TimeoutInMilliseconds $TimeoutInMilliseconds -IntervalInMilliseconds $IntervalInMilliseconds
+        $null = $ps.BeginStop($null, $null)
+        Wait-UntilTrue { $ps.InvocationStateInfo.State -eq "Stopped" } -TimeoutInMilliseconds $TimeoutInMilliseconds -IntervalInMilliseconds $IntervalInMilliseconds
+
+        $ps.InvocationStateInfo.State | Should -Be "Stopped"
```

Comment:
As this is a general purpose helper, I'm wondering whether this should be `| Should -Not -Be "Running"`
some of the states might be "failed" or "disconnected".

Reply:
If the stopping doesn't work we can get "Completed".